components/Button.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+
+interface ButtonProps {
+  label?: string;
+  children?: React.ReactNode;
+  onClick?: () => void;
+  disabled?: boolean;
+  color?: string;
+  backgroundColor?: string;
+  borderRadius?: string;
+  href?: string;
+  className?: string;
+  hoverColor?: string;
+  padding?: [string, string];
+  margin?: [string, string];
+}
+
+const Button: React.FC<ButtonProps> = ({
+  label,
+  children,
+  onClick,
+  disabled = false,
+  color = 'text-white',
+  backgroundColor = 'bg-color-1',
+  borderRadius = 'rounded-md',
+  href,
+  className,
+  hoverColor = 'hover:bg-hover-1',
+  padding = ['px-0', 'py-0'],
+  margin = ['mx-0', 'my-0'],
+}) => {
+  // Runtime validation to ensure only label or children is provided
+  if ((label && children) || (!label && !children)) {
+    throw new Error(
+      'Button component must have either a label or children, but not both.'
+    );
+  }
+
+  // Destructure padding and margin tuples
+  const [paddingX, paddingY] = padding;
+  const [marginX, marginY] = margin;
+
+  // Button content
+  const content = (
+    <span
+      className={`${color} ${backgroundColor} ${borderRadius} ${hoverColor} ${paddingX} ${paddingY} ${marginX} ${marginY}`}
+    >
+      {label || children}
+    </span>
+  );
+
+  return href ? (
+    <a
```

Comment:
instead of using Link here can we use the [Next Link Component](https://nextjs.org/docs/pages/api-reference/components/link) here it does some really cool preloading stuff, bassically when you hover over the link it loads the page content and when you click it, it replaces the vdom which prevents content flashing and page loads providng a much better user experience.

Reply:
Fixed